### PR TITLE
Add bubble sort

### PR DESCRIPTION
**Pull request by alexsaaf**
Not a very useful algorithm, but having it around doesn't hurt.

### CODE REVIEW

**Comment by alexsaaf** [comment]
Mistake there on my part, didn't realize that pushing to master would add it to the pull request. Does anyone know of a simple way to get rid of those commits?

**Comment by holmgr** [comment]
You could do an interactive rebase and then force push. Or a reset --hard then force push. This will rewirte the history so be careful

File: src/sorting.cpp [hunk 1]
```diff
@@ -133,3 +133,45 @@ std::vector<int> counting_sort(std::vector<int>& vec, int k) {
 
   return result;
 }
+
+void bubble_sort(std::vector<int>& vec) {
```
Comment (holmgr) [inline src/sorting.cpp]: I would like to see some sort of function level comment in the style of the other algorithms already implemeneted

File: src/sorting.cpp [hunk 2]
```diff
@@ -133,3 +133,45 @@ std::vector<int> counting_sort(std::vector<int>& vec, int k) {
 
   return result;
 }
+
```
Comment (fumoffu947) [inline src/sorting.cpp:136]: I would like to see a litle explenation of the function befor the implementation as the other sorting-funtions have. With just maby som information about the sort, (as this repo is for learning, but it is not nesserary)

File: src/sorting.cpp [hunk 3]
```diff
@@ -133,3 +133,45 @@ std::vector<int> counting_sort(std::vector<int>& vec, int k) {
...
+        swapped = true;
+        // Swap the elements
+        int tmp = vec[i - 1];
+        vec[i - 1] = vec[i];
+        vec[i] = tmp;
+      }
+    }
+  } while (swapped);  // If we did not swap anyting, the list is sorted
+}
+
```
Comment (fumoffu947) [inline src/sorting.cpp]: I would like to see a litle explenation of the function befor the implementation as the other sorting-funtions have. With just maby som information about the sort, (as this repo is for learning, but it is not nesserary)

**Review by fumoffu947 — COMMENTED** [review]
I would like to see the change so the repo will learn the users.

File: src/sorting.cpp (hunk 2 above)
Comment (alexsaaf) [reply src/sorting.cpp:136]: I decided to skip the description as we have started adding info to Readme.md. Maybe we should still have descriptions in the above the functions? @holmgr @fumoffu947 @Adzox

Comment (fumoffu947) [reply src/sorting.cpp:136]: I dont know what is best, but should we not have descriptions??
It really depends on the purpose of the repo. If it is for learning the algorithm then a small description would be nice, like the complexity of the function is nice to have. @holmgr @alexsaaf @Adzox

**Review by fumoffu947 — APPROVED** [review]

**Review by holmgr — APPROVED** [review]
Looks good to me, I agree that having a short description of the function is a generally good thing to have

**Pull request merged**